thetis/diagnostics.py

```diff
@@ -140,10 +154,36 @@ def __init__(self, field_2d, hessian_2d, gradient_2d=None, **kwargs):
         prob = LinearVariationalProblem(a, L, sol)
         kwargs.setdefault('solver_parameters', sp)
         self.solver = LinearVariationalSolver(prob, **kwargs)
+        self._isfrozen = True
 
     @PETSc.Log.EventDecorator("thetis.HessianRecoverer2D.solve")
     def solve(self):
         self.solver.solve()
         self.hessian_2d.assign(self._hessian)
         if self.gradient_2d is not None:
             self.gradient_2d.assign(self._gradient)
+
+
+class DynamicPressureCalculator(FrozenHasTraits):
```

Comment:
Note that there's already is a version implemented as a `DiagnosticCallback` https://github.com/thetisproject/thetis/blob/master/examples/freshwaterCylinder/freshwaterCylinder.py#L124
It could use the Calculator internally.

Reply:
Good idea. That would also mean we test the new calculator.